.github/workflows/tests.yml

```diff
@@ -272,6 +272,110 @@ jobs:
+
+      - name: Upload failed bottles
+        if: always() && steps.bottles.outputs.failures > 0
+        uses: actions/upload-artifact@main
+        with:
+          name: bottles-${{ matrix.runner }}
+          path: ${{matrix.workdir || github.workspace}}/bottles/failed
+
+      # Must be run before the `Upload bottles` step so that failed
+      # bottles are not included in the `bottles` artifact.
+      - name: Delete failed bottles
+        if: always()
+        run: rm -rvf bottles/failed
+
+      - name: Upload bottles
+        if: always() && steps.bottles.outputs.count > 0
+        uses: actions/upload-artifact@main
+        with:
+          name: bottles
+          path: ${{matrix.workdir || github.workspace}}/bottles
+
+      - name: Post cleanup
+        if: always()
+        run: |
+          brew test-bot --only-cleanup-after
+          rm -rvf bottles
+
+  test_deps:
+    needs: tests
+    if: ${{github.event_name == 'pull_request' && fromJson(needs.setup_tests.outputs.syntax-only) == false && fromJson(needs.setup_tests.outputs.test-dependents)}}
+    strategy:
+      matrix:
+        include:
+          - runner: "13-arm64-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "13-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "12-arm64"
+          - runner: "12-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "11-arm64"
+          - runner: "11-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: ${{needs.setup_tests.outputs.linux-runner}}
+            container:
+              image: ghcr.io/homebrew/ubuntu22.04:master
+              options: --user=linuxbrew -e GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED
+            workdir: /github/home
+            timeout: 4320
+      fail-fast: ${{fromJson(needs.setup_tests.outputs.fail-fast)}}
+    runs-on: ${{matrix.runner}}
+    container: ${{matrix.container}}
+    timeout-minutes: ${{ matrix.timeout || fromJson(needs.setup_tests.outputs.timeout-minutes) }}
+    defaults:
+      run:
+        shell: /bin/bash -e {0}
+        working-directory: ${{matrix.workdir || github.workspace}}
+    env:
+      GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      HOMEBREW_GITHUB_API_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    steps:
+      - name: Set environment variables
+        if: runner.os == 'macOS'
+        run: |
+          echo 'PATH=/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin' >> $GITHUB_ENV
+
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Enable debug mode
+        run: |
+          echo 'HOMEBREW_DEBUG=1' >> "${GITHUB_ENV}"
+          echo 'HOMEBREW_VERBOSE=1' >> "${GITHUB_ENV}"
+        if: runner.debug
+
+      - run: brew test-bot --only-cleanup-before
+
+      - name: Download bottles from GitHub Actions
+        uses: actions/download-artifact@v3
+        with:
+          name: bottles
+          path: ~/bottles/
```

Comment:
Does this need changes in `test-bot` or `brew` to make use of the just-downloaded bottles?

I've been thinking about this a little, and I think we might want to add a (perhaps hidden) flag `--prefer-local-bottles` to `brew install` which makes `brew install` check the working directory first for a possible bottle to use before trying to download one.

The reason this should be in `brew install` rather than `brew test-bot` is that any given `brew install` operation might install dependencies too -- and the dependency resolution logic lives in `brew install` and not `brew test-bot`.

Reply:
@carlocab Yes, it feels like it will need that change. Before I merge something like this I was gonna test this out with bottles in this PR anyway.

Reply:
Implementing in https://github.com/Homebrew/homebrew-test-bot/pull/886

Reply:
Implementing in https://github.com/Homebrew/homebrew-test-bot/pull/886